File: kill-process.rb
#!/usr/bin/env ruby

# Search for process to kill with Rofi

data = `ps --user "$(id -u)" -o pid,time,cmd | rofi -dmenu -i | xargs -r echo`.strip.split(" ")

if data.size >= 3 && data[0] != "PID"
    pid = data[0]

    `kill -9 "#{pid}"`
    puts "Killing process with pid=#{pid}"
end

File: kill-process.clj
#!/usr/bin/env bb

; Shows list of process with Rofi and kills selected

(require '[clojure.java.shell :refer [sh]]
         '[clojure.string :as str])

(defn run [command]
  (let [result (sh "bash" "-c" command)]
    (if (contains? result :out) (:out result) nil)))

(def pid (str/trim (run "ps --user \"$(id -u)\" -o pid,time,cmd | rofi -dmenu -i | awk '{print $1}'")))

(if (and (some? pid) (not= pid "PID"))
  (do 
    (println (format "Killing process %s" pid))
    (run (format "kill -9 %s" pid))
    nil))
